### PR TITLE
WEBUI-247: add flag column

### PR DESCRIPTION
**Pull request by andreacornaglia**

### CODE REVIEW

File: elements/nuxeo-results/nuxeo-default-results.js [hunk 1]
```diff
@@ -212,6 +212,21 @@ Polymer({
             </template>
           </template>
         </nuxeo-data-table-column>
+        <nuxeo-data-table-column name="[[i18n('documentContentView.datatable.header.flags')]]" flex="50">
```
Comment (Gabez0r) [inline elements/nuxeo-results/nuxeo-default-results.js:215]: I think this column should be hidden by default, as most users will just get an empty column, which will take a lot of space.

File: elements/nuxeo-results/nuxeo-document-content.js [hunk 2]
```diff
@@ -269,6 +274,21 @@ Polymer({
             </template>
           </template>
         </nuxeo-data-table-column>
+        <nuxeo-data-table-column name="[[i18n('documentContentView.datatable.header.flags')]]" flex="50">
```
Comment (Gabez0r) [inline elements/nuxeo-results/nuxeo-document-content.js:277]: Same here.

File: elements/nuxeo-results/nuxeo-default-results.js [hunk 3]
```diff
@@ -212,6 +212,21 @@ Polymer({
...
+              <iron-icon id="retainIcon" icon="nuxeo:retain"></iron-icon>
+              <nuxeo-tooltip for="retainIcon">[[i18n('documentContentView.datatable.flags.retention')]]</nuxeo-tooltip>
+            </template>
+            <template is="dom-if" if="[[item.hasLegalHold]]">
+              <iron-icon id="legalHoldIcon" icon="nuxeo:hold"></iron-icon>
+              <nuxeo-tooltip for="legalHoldIcon"
+                >[[i18n('documentContentView.datatable.flags.legalHold')]]</nuxeo-tooltip
+              >
+            </template>
+            <nuxeo-slot slot="ADDITIONAL_FLAGS"></nuxeo-slot>
```
Comment (Gabez0r) [inline elements/nuxeo-results/nuxeo-default-results.js:227]: Do we have a requirement to expose this slot?

File: elements/nuxeo-results/nuxeo-default-results.js (hunk 1 above)
Comment (andreacornaglia) [reply elements/nuxeo-results/nuxeo-default-results.js]: done

File: elements/nuxeo-results/nuxeo-default-results.js (hunk 3 above)
Comment (andreacornaglia) [reply elements/nuxeo-results/nuxeo-default-results.js]: not directly so i'll remove it for now.

**Review by Gabez0r — COMMENTED** [review]
In what elements are the flags supposed to be visible? Right now, this will only make them available on document content and collections, but it won't show on search results, for example. Is it supposed? Can we please document on the ticket in which elements we want to make this available?

**Comment by andreacornaglia** [comment]
> In what elements are the flags supposed to be visible? Right now, this will only make them available on document content and collections, but it won't show on search results, for example. Is it supposed? Can we please document on the ticket in which elements we want to make this available?

@Gabez0r inded, I'm sorry about that. The goal is to make it available in search results and document list, I targeted by mistake the wrong component (nuxeo-default-results), now it should be ok.

**Review by richardsd — DISMISSED** [review]

File: elements/nuxeo-results/nuxeo-document-content.js [hunk 4]
```diff
@@ -30,6 +30,7 @@ import '../nuxeo-data-grid/nuxeo-document-grid-thumbnail.js';
 import './nuxeo-results.js';
 import { Polymer } from '@polymer/polymer/lib/legacy/polymer-fn.js';
 import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { FiltersBehavior } from '@nuxeo/nuxeo-ui-elements//nuxeo-filters-behavior.js';
```
Comment (mnixo) [inline elements/nuxeo-results/nuxeo-document-content.js:33]:
```suggestion
import { FiltersBehavior } from '@nuxeo/nuxeo-ui-elements/nuxeo-filters-behavior.js';
```

**Review by mnixo — APPROVED** [review]

File: elements/nuxeo-results/nuxeo-document-content.js (hunk 4 above)
Comment (andreacornaglia) [reply elements/nuxeo-results/nuxeo-document-content.js]: ops. thanks for the catch, i removed the extra slash.

**Review by mnixo — DISMISSED** [review]

File: elements/search/default/nuxeo-default-search-results.html [hunk 5]
```diff
@@ -158,6 +163,32 @@
...
+            <template is="dom-if" if="[[item.hasLegalHold]]">
+              <iron-icon id="legalHoldIcon" icon="nuxeo:hold"></iron-icon>
+              <nuxeo-tooltip for="legalHoldIcon">
+                [[i18n('documentContentView.datatable.flags.legalHold')]]
+              </nuxeo-tooltip>
+            </template>
+            <template is="dom-if" if="[[isFavorite(item)]]">
+              <iron-icon id="favorite" icon="nuxeo:favorites"></iron-icon>
+              <nuxeo-tooltip for="favorite">
+                [[i18n('documentContentView.datatable.flags.favorite')]]
```
Comment (Gabez0r) [inline elements/search/default/nuxeo-default-search-results.html:181]: We are missing this label in messages.json.

**Review by Gabez0r — COMMENTED** [review]
@andreacornaglia, looking good, thank you. But we still need document where these flags should be visible exactly. Could you please check with Lisa and Bertrand if it is also supposed to be visible in:
- collections
- trash content of a document
- trash search
- asset search
- expired search

These cases are currently not covered but they are still considered "document listing" and "search results".

File: elements/search/default/nuxeo-default-search-results.html (hunk 5 above)
Comment (andreacornaglia) [reply elements/search/default/nuxeo-default-search-results.html:181]: good catch! i added it. thanks!

**Comment by andreacornaglia** [comment]
> @andreacornaglia, looking good, thank you. But we still need document where these flags should be visible exactly. Could you please check with Lisa and Bertrand if it is also supposed to be visible in:
> 
> * collections
> * trash content of a document
> * trash search
> * asset search
> * expired search
> 
> These cases are currently not covered but they are still considered "document listing" and "search results".

@Gabez0r makes sense! Thanks for bringing this up. I did talk to Lisa and she replied directly on the ticket, but in summary, she agrees the flags could be available on any document listing. I will look into those cases and add them accordingly.

**Review by mnixo — APPROVED** [review]

**Review by mnixo — DISMISSED** [review]

File: i18n/messages.json [hunk 6]
```diff
@@ -362,6 +363,10 @@
   "documentContentView.datatable.header.title": "Title",
   "documentContentView.datatable.header.type": "Type",
   "documentContentView.datatable.header.version": "Version",
+  "documentContentView.datatable.flags.retention": "Under retention",
+  "documentContentView.datatable.flags.legalHold": "Under legal hold",
+  "documentContentView.datatable.flags.coldStorage": "Under cold storage",
+  "documentContentView.datatable.flags.favorite": "Under favorites",
```
Comment (semisse) [inline i18n/messages.json:369]: we should move this above since it is not alphabetically ordered

Comment (andreacornaglia) [reply i18n/messages.json]: ok, done

**Review by mnixo — DISMISSED** [review]

File: i18n/messages.json [hunk 7]
```diff
@@ -362,6 +363,10 @@
   "documentContentView.datatable.header.title": "Title",
   "documentContentView.datatable.header.type": "Type",
   "documentContentView.datatable.header.version": "Version",
+  "documentContentView.datatable.flags.coldStorage": "Under cold storage",
+  "documentContentView.datatable.flags.favorite": "Under favorites",
+  "documentContentView.datatable.flags.legalHold": "Under legal hold",
+  "documentContentView.datatable.flags.retention": "Under retention",
```
Comment (semisse) [inline i18n/messages.json:369]: You forgot to order these ones as well 😅

**Review by semisse — APPROVED** [review]

**Review by mnixo — APPROVED** [review]

**Pull request merged**